woodwork/column_accessor.py

```diff
@@ -103,7 +104,10 @@ def init(
                     logical_type.validate(self._series)
                 else:
                     valid_dtype = logical_type._get_valid_dtype(type(self._series))
-                    if valid_dtype != str(self._series.dtype):
+                    if valid_dtype != str(self._series.dtype) and not (
+                        pdtypes.is_integer_dtype(valid_dtype)
+                        and pdtypes.is_float_dtype(self._series.dtype)
+                    ):
```

Comment:
Needed because `nans` will currently render an otherwise `integer` dtype as a `float`. Conversion to `Int64` dtype then yields an error